app/models/form_profiles/va21686c.rb

```diff
@@ -1,11 +1,249 @@
+    VA21686c::FormContactInformation.new(
+      {
+        veteran_address: prefill_address(veteran['address']),
+        veteran_full_name: prefill_name(veteran),
+        veteran_email: veteran['emailAddress'],
+        va_file_number: detect_file_number(veteran['vaFileNumber']),
+        marital_status: veteran['marriageType'],
+        day_phone: convert_phone(veteran['primaryPhone']),
+        night_phone: convert_phone(veteran['secondaryPhone']),
+        veteran_social_security_number: convert_ssn(veteran['ssn']),
+        current_marriage: prefill_current_marriage(veteran['spouse']),
+        previous_marriages: veteran['previousMarriages'].map { |m| prefill_marriage(m) },
+        dependents: prefill_dependents(veteran['children'])
+      }.compact
+    )
+  end
+
+  def convert_ssn(ssn)
+    return unless ssn
+    ssn.tr('^0-9', '')
+  end
+
+  def convert_phone(phone)
+    return unless phone
+    "#{phone['areaNbr']}#{phone['phoneNbr']}".tr('^0-9', '')
+  end
+
+  def prefill_current_marriage(spouse)
+    return unless spouse
+    marriage = spouse['currentMarriage']
+    VA21686c::FormCurrentMarriage.new(
+      {
+        date_of_marriage: convert_date(marriage['marriageDate']),
+        location_of_marriage: prefill_location(marriage['country'], marriage['city'], marriage['state']),
+        spouse_full_name: prefill_name(spouse),
+        spouse_social_security_number: convert_ssn(spouse['ssn']),
+        spouse_has_no_ssn: spouse['hasNoSsn'],
+        spouse_marriages: spouse['previousMarriages'].map { |m| prefill_marriage(m) },
+        spouse_address: prefill_address(spouse['address']),
+        spouse_is_veteran: spouse['veteran'],
+        live_with_spouse: spouse['sameResidency'],
+        spouse_date_of_birth: convert_date(spouse['dateOfBirth'])
+      }.compact
+    )
+  end
+
+  def detect_file_number(file_number)
+    return if file_number.nil? || file_number.match(/^\d{3}-\d{2}-\d{4}$/)
+    file_number
+  end
+
+  def prefill_marriage(marriage)
+    return unless marriage
+    VA21686c::FormMarriage.new(
+      {
+        date_of_marriage: convert_date(marriage['marriageDate']), # ???
+        location_of_marriage: prefill_location(marriage['country'], marriage['city'], marriage['state']),
+        spouse_full_name: prefill_name(marriage),
+        reason_for_separation: marriage['marriageTerminationReasonType'],
+        date_of_separation: convert_date(marriage['terminatedDate']),
+        location_of_separation: prefill_location(marriage['endCountry'], marriage['endCity'], marriage['endState'])
+      }.compact
+    )
+  end
+
+  def prefill_location(country, city, state)
+    country ||= {}
+    VA21686c::FormLocation.new(
+      {
+        country_dropdown: country['dropDownCountry'],
+        country_text: country['textCountry'],
+        city: city,
+        state: state
+      }.compact
+    )
+  end
+
+  def convert_date(date)
+    return unless date
+    Date.strptime(date.to_s, '%Q').strftime('%Y-%m-%d')
```

Comment:
Is this an appropriate date format to return?

Reply:
I checked, and the format matches other date examples in `form_profile_spec`

Reply:
can you change `strftime('%Y-%m-%d')` to `to_s`?